tools/test/h5copy/CMakeTests.cmake

```diff
@@ -58,215 +100,328 @@
   #
   # Perform h5copy according to passing parameters
   #
-  macro (ADD_H5_F_TEST testname resultcode infile fparam vparam sparam srcname dparam dstname)
+  macro (ADD_H5_F_TEST testname vol env resultcode infile fparam vparam sparam srcname dparam dstname)
     # Remove any output file left over from previous test run
     add_test (
-        NAME H5COPY_F-${testname}-clear-objects
-        COMMAND ${CMAKE_COMMAND} -E remove ./testfiles/${testname}.out.h5
+        NAME HDF5_VOL_${vol}-H5COPY_F-${testname}-clear-objects
+        COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:h5delete> ./testfiles/${testname}.out.h5
     )
 
+    set_tests_properties (HDF5_VOL_${vol}-H5COPY_F-${testname}-clear-objects PROPERTIES
+      ENVIRONMENT "${env}"
```

Comment:
Similarly, we should probably avoid modifying any of these properties when using the native VOL. Currently, this seems like it would pass NONE for the environment, whereas we shouldn't be passing anything. It would also be good to avoid the generator expressions below that are trying to fix things up when using the native VOL.